04-onion-routing.md

```diff
@@ -73,42 +72,42 @@ A node:
 
 There are a number of conventions adhered to throughout this document:
 
- - Length: the maximum route length is limited to 20 hops.
  - HMAC: the integrity verification of the packet is based on Keyed-Hash
- Message Authentication Code, as defined by the
- [FIPS 198 Standard](http://csrc.nist.gov/publications/fips/fips198-1/FIPS-198-1_final.pdf)/[RFC 2104](https://tools.ietf.org/html/rfc2104),
- and using a `SHA256` hashing algorithm.
- - Elliptic curve: for all computations involving elliptic curves, the
-   Bitcoin curve is used, as specified in [`secp256k1`](http://www.secg.org/sec2-v2.pdf).
- - Pseudo-random stream: [`ChaCha20`](https://tools.ietf.org/html/rfc7539) is
- used to generate a pseudo-random byte stream. For its generation, a fixed
- null-nonce (`0x0000000000000000`) is used, along with a key derived from a
- shared secret and with a `0x00`-byte stream of the desired output size as the
- message.
-
+   Message Authentication Code, as defined by the [FIPS 198
+   Standard][fips198]/[RFC 2104][RFC2104], and using a `SHA256` hashing
+   algorithm.
+ - Elliptic curve: for all computations involving elliptic curves, the Bitcoin
+   curve is used, as specified in [`secp256k1`][sec2]
+ - Pseudo-random stream: [`ChaCha20`][rfc7539] is used to generate a
+   pseudo-random byte stream. For its generation, a fixed null-nonce
+   (`0x0000000000000000`) is used, along with a key derived from a shared
+   secret and with a `0x00`-byte stream of the desired output size as the
+   message.
  - The terms _origin node_ and _final node_ refer to the initial packet sender
- and the final packet recipient, respectively.
+   and the final packet recipient, respectively.
  - The terms _hop_ and _node_ are sometimes used interchangeably, but a _hop_
- usually refers to an intermediate node in the route rather than an end node.
+   usually refers to an intermediate node in the route rather than an end node.
         _origin node_ --> _hop_ --> ... --> _hop_ --> _final node_
  - The term _processing node_ refers to the specific node along the route that is
- currently processing the forwarded packet.
+   currently processing the forwarded packet.
  - The term _peers_ refers only to hops that are direct neighbors (in the
- overlay network): more specifically, _sending peers_ forward packets to
- _receiving peers_.
-        _sending peer_ --> _receiving peer_
-
-
-# Clarifications
-
-The longest route supported has 20 hops without counting the _origin node_ and _final node_, thus 19 _intermediate nodes_ and a maximum of 20 channels to be traversed.
+   overlay network): more specifically, _sending peers_ forward packets
+   to _receiving peers_.
+ - Each hop in the route has a variable length `hop_payload`, or a fixed-size
+   legacy `hop_data` payload.
+    - The legacy `hop_data` is identified by a single `0x00`-byte prefix
```

Comment:
Is this prefix in _addition_ to the leading `0x00` realm byte?

Reply:
I think the wording here is simply getting rid of the `realm` concept.
This leading `0x00` byte prefix is what we called `realm` in the previous version (don't add an additional one otherwise it's not backwards-compatible anymore).

Reply:
Yep, there is only one `0x00` byte, which used to be called `realm`, and now is just the length encoded as a varint.